### PR TITLE
Installing type_traits header file

### PR DESCRIPTION
**Pull request by mmigdal-nv**
@zasdfgbnm  is this okay? This is now needed for out-of-tree builds. Pattern matching can't work

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -248,7 +248,9 @@ endif()
 # installing nvfuser headers
 install(DIRECTORY "${NVFUSER_SRCS_DIR}/"
       DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/nvfuser"
-      FILES_MATCHING PATTERN "*.h" )
+      FILES_MATCHING
+      PATTERN "*.h"
+      PATTERN "type_traits")
```
Comment (zasdfgbnm) [inline CMakeLists.txt:253]: I think it is fine. Should we be more specific and say `csrc/C++20/type_traits` instead of  `type_traits`?

Comment (jjsjann123) [reply CMakeLists.txt:253]: Can we also add this here:
https://github.com/NVIDIA/Fuser/blob/eabfc2b0701e076c8e1530894019cd8c0cd0cf8c/setup.py#L318-L326

We probably need a test coverage on that as well. But let's not worry about that now.

**Review by jjsjann123 — CHANGES_REQUESTED** [review]

**Review by zasdfgbnm — COMMENTED** [review]
LGTM, will leave it to @jjsjann123 for approval

**Review by jjsjann123 — APPROVED** [review]
LGTM.

**Comment by jjsjann123** [comment]
Still thinking about test coverage. I'll open an issue to track that. Could be a good starter task for our release engineer.

**Pull request merged**